modules/core/test/test_intrin_utils.hpp

```diff
@@ -1139,6 +1139,37 @@ template<typename R> struct TheTest
         return *this;
     }
 
+    TheTest & test_permute()
+    {
+#if CV_SIMD_PERMUTE > 0
+        Data<R> dataA;
+        Data<R> dataB;
+
+        // We want 0 - nlanes, n
+        for (int i = 0; i < v_uint8::nlanes; i++ )
+            dataB.d[v_uint8::nlanes - 1 - i] = (LaneType)i;
+
+        // Use permute to reverse data
+        R a = v_permute(v_uint8(dataB), v_uint8(dataA));
```

Comment:
I think it make sense to include test for 2 vector permute as well to ensure compatibility if/when multi vector permute support will be extended to other platforms